src/common/quantile.cuh

```diff
@@ -16,6 +16,19 @@ class HistogramCuts;
 using WQSketch = WQuantileSketch<bst_float, bst_float>;
 using SketchEntry = WQSketch::Entry;
 
+namespace detail {
+struct IsCatOp {
+  XGBOOST_DEVICE bool operator()(FeatureType ft) {
+    return ft == FeatureType::kCategorical;
+  }
+};
+struct SketchUnique {
+  XGBOOST_DEVICE bool operator()(SketchEntry const& a, SketchEntry const& b) const {
+    return a.value - b.value == 0;
```

Comment:
What is this comparator doing?

Reply:
See if they are equal.

Reply:
Why not `a.value == b.value`?

Reply:
If memory is correct, I think that is not guaranteed to work: https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html